### PR TITLE
Added User Registration

### PR DESCRIPTION
**Pull request by OmSingh5092**
Implemented user registration by binding the APIs.

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -1,6 +1,8 @@
 <?xml version="1.0" encoding="utf-8"?>
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
     package="com.antailbaxt3r.collegemate">
+    <uses-permission android:name="android.permission.INTERNET" />
+
```
Comment (antailbaxt3r) [inline app/src/main/AndroidManifest.xml:5]: Remove unnecessary white spaces

File: app/src/main/java/com/antailbaxt3r/collegemate/activities/LoginActivity.java [hunk 2]
```diff
@@ -124,16 +128,20 @@ private void handleSignInResult(GoogleSignInResult result) {
         }
     }
 
-    private void firebaseAuthWithGoogle(AuthCredential credential) {
+    private void firebaseAuthWithGoogle(String idToken) {
+        AuthCredential credential = GoogleAuthProvider.getCredential(idToken, null);
```
Comment (antailbaxt3r) [inline app/src/main/java/com/antailbaxt3r/collegemate/activities/LoginActivity.java:133]: Don't change this,

File: app/src/main/java/com/antailbaxt3r/collegemate/activities/LoginActivity.java [hunk 3]
```diff
@@ -124,16 +128,20 @@ private void handleSignInResult(GoogleSignInResult result) {
         }
     }
 
-    private void firebaseAuthWithGoogle(AuthCredential credential) {
```
Comment (antailbaxt3r) [inline app/src/main/java/com/antailbaxt3r/collegemate/activities/LoginActivity.java:127]: This is how it is supposed to be done

File: app/src/main/java/com/antailbaxt3r/collegemate/activities/LoginActivity.java [hunk 4]
```diff
@@ -146,18 +154,32 @@ public void onComplete(@NonNull Task<AuthResult> task) {
                 });
     }
 
+    private void oAuthVerification(String idtoken){
+        APIInterface apiInterface = RetrofitClient.getClient();
+        apiInterface.sendRegToken(idtoken).enqueue(new Callback<String>() {
+            @Override
```
Comment (antailbaxt3r) [inline app/src/main/java/com/antailbaxt3r/collegemate/activities/LoginActivity.java]: Don't create apiInterface.
Use something like this:
```Call<String> call = RetrifitClient.getClient().sendRegToken(idToken);```

File: app/src/main/java/com/antailbaxt3r/collegemate/activities/LoginActivity.java [hunk 5]
```diff
@@ -146,18 +154,32 @@ public void onComplete(@NonNull Task<AuthResult> task) {
...
         intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
         startActivity(intent);
-        finish();
     }
 
-    private void goToOnboarding() {
-        Intent intent = new Intent(LoginActivity.this, MainActivity.class);
-        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
-        startActivity(intent);
-        finish();
```
Comment (antailbaxt3r) [inline app/src/main/java/com/antailbaxt3r/collegemate/activities/LoginActivity.java:160]: Don't remove this. This is fine

File: app/src/main/java/com/antailbaxt3r/collegemate/retrofit/APIInterface.java [hunk 6]
```diff
@@ -1,4 +1,20 @@
...
+
 public interface APIInterface {
+    @POST("api/user/update")
+    Call<ProfileModal> createProfile(@Body ProfileModal profileModal);
+
+    @POST("api/google/signin")
+    @FormUrlEncoded
+    Call<String> sendRegToken(@Field("id_token") String idtoken);
+
+
```
Comment (antailbaxt3r) [inline app/src/main/java/com/antailbaxt3r/collegemate/retrofit/APIInterface.java]: Don't get String, get Model

File: app/src/main/res/layout/activity_register.xml [hunk 7]
```diff
@@ -34,65 +36,68 @@
...
-                android:background="@drawable/rounded_rectangle_10dp"
-                android:padding="10dp"
                 android:layout_margin="20dp"
-                android:hint="Enrollment Id"
+                android:background="@drawable/rounded_rectangle_10dp"
                 android:fontFamily="@font/raleway_reg"
-                android:textColor="@color/dark_blue"/>
+                android:hint="Enrollment Id"
+                android:padding="10dp"
+                android:textColor="@color/dark_blue" />
```
Comment (antailbaxt3r) [inline app/src/main/res/layout/activity_register.xml:57]: Revert these changes, unnecessary

**Review by antailbaxt3r — CHANGES_REQUESTED** [review]

File: app/src/main/res/layout/activity_register.xml (hunk 7 above)
Comment (OmSingh5092) [reply app/src/main/res/layout/activity_register.xml:57]: This seems wrong to me.

File: app/src/main/java/com/antailbaxt3r/collegemate/activities/LoginActivity.java (hunk 5 above)
Comment (OmSingh5092) [reply app/src/main/java/com/antailbaxt3r/collegemate/activities/LoginActivity.java:160]: Two Functions with same functionality was present. Seemed unnecessary.

File: app/src/main/java/com/antailbaxt3r/collegemate/activities/LoginActivity.java (hunk 2 above)
Comment (OmSingh5092) [reply app/src/main/java/com/antailbaxt3r/collegemate/activities/LoginActivity.java:133]: Needed this to implement OAuth verification function.

File: app/src/main/java/com/antailbaxt3r/collegemate/activities/LoginActivity.java (hunk 3 above)
Comment (OmSingh5092) [reply app/src/main/java/com/antailbaxt3r/collegemate/activities/LoginActivity.java:127]: I have written the code according to Firebase documentation.

**Comment by antailbaxt3r** [comment]
Closing this for example code
